imap_processing/swapi/l1/swapi_l1.py

```diff
@@ -1 +1,565 @@
+        COIN_CNT_5   -> [x, x, x, x, x,..., x, x, x, ..., x, x, x]
+
+    This function reads one full sweep data in this order:
+        PCEM_CNT0 --> 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11
+        PCEM_CNT1 --> 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11
+        PCEM_CNT2 --> 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11
+        PCEM_CNT3 --> 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11
+        PCEM_CNT4 --> 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11
+        PCEM_CNT5 --> 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11
+
+    This example show for PCEM_CNT but same logic applies
+    to SCEM_CNT, COIN_CNT, PCEM_RNG, SCEM_RNG, and COIN_RNG.
+
+    In the final L1A product of 1x72 array where we store
+    final PCEM, SCEM, COIN counts or compression indicator
+    such as PCEM_RNG, SCEM_RNG, COIN_RNG,
+    we want data in this order. Transpose of above layout
+    0, 0, 0, 0, 0, 0,
+    1, 1, 1, 1, 1, 1,
+    2, 2, 2, 2, 2, 2,
+    3, 3, 3, 3, 3, 3,
+    ....,
+    11, 11, 11, 11, 11, 11.
+    Reordering in this order is reordering all data of
+    sequence 0 first, then sequence 1, then sequence 2,
+    and so on until sequence 11.
+
+    Parameters
+    ----------
+    full_sweep_sci : xarray.Dataset
+        Science data that only contains full sweep data.
+    sweep_index : int
+        Start index of current sweep.
+
+    Returns
+    -------
+    pcem_count : list
+        List of PCEM counts.
+    scem_count : list
+        List of SCEM counts.
+    coin_count : list
+        List of COIN counts.
+    pcem_rng_val : list
+        List of PCEM compression indicator.
+    scem_rng_val : list
+        List of SCEM compression indicator.
+    coin_rng_val : list
+        List of COIN compression indicator.
+    """
+    # Arrays to store 1x72 values
+    pcem_count = []
+    scem_count = []
+    coin_count = []
+    pcem_rng_val = []
+    scem_rng_val = []
+    coin_rng_val = []
+    seq_num = []
+
+    m = sweep_index
+    n = sweep_index + 12
+
+    # This for loop is going through each data field.
+    # For example,
+    # at range 0, we get one full sweep data from
+    # PCEM_CNT0, SCEM_CNT0, COIN_CNT0, PCEM_RNG_ST0, SCEM_RNG_ST0
+    # COIN_RNG_ST0
+    # at range 1, we get one full sweep data from
+    # PCEM_CNT1, SCEM_CNT1, COIN_CNT1, PCEM_RNG_ST1, SCEM_RNG_ST1
+    # ...
+    # at range 5, we get one full sweep data from
+    # PCEM_CNT5, SCEM_CNT5, COIN_CNT5, PCEM_RNG_ST5, SCEM_RNG_ST5
+    for step in range(6):
+        seq_num.append(list(full_sweep_sci["SEQ_NUMBER"].data[m:n][:]))
+        # Using list(value) to convert xarray.DataArray to list
+        # for easier manipulation later
+
+        # For each step, read full sweep data for each
+        # PCEM, SCEM, and COIN. Uncompressed counts as needed
+        pcem_raw_count = list(full_sweep_sci[f"PCEM_CNT{step}"].data[m:n])
+        scem_raw_count = list(full_sweep_sci[f"SCEM_CNT{step}"].data[m:n])
+        coin_raw_count = list(full_sweep_sci[f"COIN_CNT{step}"].data[m:n])
```

Comment:
Can you create empty numpy arrays above and fill in the values rather than appending and expanding a list which is slower? Then I think at the end you could do `arr.T` to transpose your array if you need as well.

Reading through this again, I might even look into whether going to one array of shape `np.empty((6, 72))` and index in with step would work.
```suggestion
        # For each step, read full sweep data for each
        # PCEM, SCEM, and COIN. Uncompressed counts as needed
        pcem_raw_count[step::6] = full_sweep_sci[f"PCEM_CNT{step}"].data[m:n])
        # depending on how you want to store/order these I think you can mess with the stride / start-stop as you need.
```